STYLEGUIDE.md

```diff
@@ -24,3 +24,16 @@
    Guide](https://google.github.io/styleguide/shellguide.html) in mind.
    * In particular, check out the "When to use Shell" section to consider if
      it's time to switch languages.
+
+## Stack choice
+
+Most Paketo buildpacks can be divided into two categories - 
+
+- Dependency/Distribution buildpacks like - [go-dist](https://github.com/paketo-buildpacks/go-dist), [node-engine](https://github.com/paketo-buildpacks/node-engine), [yarn](https://github.com/paketo-buildpacks/yarn), [dep](https://github.com/paketo-buildpacks/dep) etc. which provide tarball distributions which maybe architecture or linux distro specific.
+- Configuration/Utility buildpacks like - [procfile](https://github.com/paketo-buildpacks/procfile), [environment-variables](https://github.com/paketo-buildpacks/environment-variables), [go-mod-vendor](https://github.com/paketo-buildpacks/go-mod-vendor), [npm-install](https://github.com/paketo-buildpacks/npm-install) etc. which use distribution buildpacks or are completely standalone.
+
+1. Whenever possible, we should set the most permissive stack id for a buildpack. The Cloud Native Buildpack [spec](https://github.com/buildpacks/spec/blob/buildpack/0.5/buildpack.md#buildpack-implementations) allows for a wildcard stack `*` which indicates compatibility with any stack. 
+
+1. The wildcard stack should be the stack of choice for utility or configuration buildpacks provided they can work on any linux stack. This should be the case for buildpacks written in `go` that are standalone and don't rely on any implicit stack dependencies.
```

Comment:
Would we not want something like `linux/*`? What about Windows? or other OS's (theoretical at the moment)? If my buildpack is compiled for Linux, it wouldn't even run on Windows to get to the point where I could do more validation about the platform. Or would that be handled outside of my buildpack before my buildpack even runs?

Reply:
OK, got it. That makes sense and keeps it simple. It probably wouldn't hurt to mention that in the doc. Other readers may have similar thoughts.